pkg/dockerregistry/server/pullthroughmanifestservice_test.go

```diff
@@ -156,11 +167,18 @@ func TestPullthroughManifests(t *testing.T) {
 			if tc.expectedNotFoundError && err == distribution.ErrBlobUnknown {
 				break
 			}
+			if e, ok := err.(*kerrors.StatusError); ok {
+				if tc.expectedNotFoundError && e.ErrStatus.Code == http.StatusNotFound {
```

Comment:
Yes, but it's wrong. StatusError will be converted to 500 unknown error. We should propagate the information to the client where possible.

If you don't want to touch that in this PR, please add at least a TODO here to rewrite kube errors to distribution errors in the middleware.